Math/NumberTheory/GCD.hs

```diff
@@ -26,14 +26,23 @@
 {-# LANGUAGE LambdaCase   #-}
 {-# LANGUAGE MagicHash    #-}
 
+{-# OPTIONS_GHC -fno-warn-unused-imports -fno-warn-unused-top-binds #-}
```

Comment:
I'd rather export `insert` outside than use `-fno-warn-unused-top-binds`. The latter does not work for GHC 7.10, causing build failure on Travis.